python/pyarrow/_dataset_parquet.pyx

```diff
@@ -56,9 +63,180 @@ from pyarrow._parquet cimport (
+            Shared pointer to a `DecryptionConfiguration` object, specifying decryption settings
+            for reading encrypted Parquet data.
+
+        Raises
+        ------
+        ValueError
+            Raised if `decryption_config` is None.
+        """
+
+        cdef:
+            shared_ptr[CParquetDecryptionConfig] c_config
+
+        # Avoid mistakingly creating attributes
+        __slots__ = ()
+
+        def __cinit__(self, CryptoFactory crypto_factory, KmsConnectionConfig kms_connection_config,
+                      DecryptionConfiguration decryption_config):
+
+            cdef shared_ptr[CDecryptionConfiguration] c_decryption_config
+
+            if decryption_config is None:
+                raise ValueError(
+                    "decryption_config cannot be None")
+
+            self.c_config.reset(new CParquetDecryptionConfig())
+
+            c_decryption_config = ParquetDecryptionConfig.unwrap_decryptionconfig(
+                decryption_config)
+
+            self.c_config.get().crypto_factory = ParquetDecryptionConfig.unwrap_cryptofactory(crypto_factory)
+            self.c_config.get().kms_connection_config = ParquetDecryptionConfig.unwrap_kmsconnectionconfig(
+                kms_connection_config)
+            self.c_config.get().decryption_config = c_decryption_config
+
+        @staticmethod
+        cdef wrap(shared_ptr[CParquetDecryptionConfig] c_config):
+            cdef ParquetDecryptionConfig python_config = ParquetDecryptionConfig.__new__(ParquetDecryptionConfig)
+            python_config.c_config = c_config
+            return python_config
+
+        cdef shared_ptr[CParquetDecryptionConfig] unwrap(self):
+            return self.c_config
+
+        @staticmethod
+        cdef shared_ptr[CCryptoFactory] unwrap_cryptofactory(object crypto_factory) except *:
+            if isinstance(crypto_factory, CryptoFactory):
+                pycf = (<CryptoFactory> crypto_factory).unwrap()
+                return static_pointer_cast[CCryptoFactory, CPyCryptoFactory](pycf)
+            raise TypeError("Expected CryptoFactory, got %s" % type(crypto_factory))
+
+        @staticmethod
+        cdef shared_ptr[CKmsConnectionConfig] unwrap_kmsconnectionconfig(object kmsconnectionconfig) except *:
+            if isinstance(kmsconnectionconfig, KmsConnectionConfig):
+                return (<KmsConnectionConfig> kmsconnectionconfig).unwrap()
+            raise TypeError("Expected KmsConnectionConfig, got %s" %
+                            type(kmsconnectionconfig))
+
+        @staticmethod
+        cdef shared_ptr[CDecryptionConfiguration] unwrap_decryptionconfig(object decryptionconfig) except *:
+            if isinstance(decryptionconfig, DecryptionConfiguration):
+                return (<DecryptionConfiguration> decryptionconfig).unwrap()
+
+            raise TypeError("Expected DecryptionConfiguration, got %s" %
+                            type(decryptionconfig))
```

Comment:
I may be mistaken, but it seems these three static methods are the same as in from `ParquetEncryptionConfig`? To avoid duplication, I think you can simply make them top-level functions instead of methods.